### PR TITLE
feat(Azure): Add Azure module

### PR DESCRIPTION
**Pull request by andymac4182**
#### Description
Replacing https://github.com/starship/starship/pull/1086

Sorry I am not 100% what needs to be changed. I will try to look later but I am still very new to Rust.

### CODE REVIEW

**Comment by andymac4182** [comment]
This is looking good to me. The test failure seems related to other modules

**Comment by andymac4182** [comment]
Any feedback on this?

File: docs/config/README.md [hunk 1]
```diff
@@ -292,6 +292,27 @@ style = "bold blue"
...
+## AZURE
+
+The `azure` module shows the current Azure Subscription. This is based on 
+`~/.azure/clouds.config` and `~/.azure/azureProfile.json` files.
+
+### Options
+
+| Variable          | Default         | Description                                                                 |
+| ----------------- | --------------- | --------------------------------------------------------------------------- |
+| `format`          | `"on ☁️ [$subscription](blue bold) "`         | The format for the Azure module to render in                  |
```
Comment (andytom) [inline docs/config/README.md:304]: With the other modules we have been keeping the symbol and style as separate variables to make it easier for users to change just the symbol or style.

**Comment by andrejohansson** [comment]
Really looking forward to this module! How often does PRs get merged?

**Comment by riweston** [comment]
Is somebody able to merge this? Keen to integrate to my terminal!

**Comment by JustinGrote** [comment]
@andytom Anything further required?

**Comment by andytom** [comment]
> @andytom Anything further required?

The PR cannot be merged at the moment because of the merge conflicts so they would need to be fixed first. I would also like to have the changes to the configuration to separate out the symbol and style from the format string so that this module is the same as the other modules giving a more consistent user experience.

**Comment by JustinGrote** [comment]
Ah my apologies, I thought those were resolved in the commits after your request. Maybe I'll take a stab at a new PR if @andymac4182 doesn't want to pursue it further

**Comment by andytom** [comment]
I am going to close this PR as it is not possible to merge it in it's current state
@andymac4182 I am sorry this did not get merged but please feel free to reopen it in the future. Also @JustinGrote if you want to take a shot at continuing the work feel free to pick this up yourself.

**Pull request closed without merging**